substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationSupport.java

```diff
@@ -106,23 +106,26 @@ public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Rec
                  * String arguments.
                  */
                 if (libnameNode.isConstant()) {
-                    String libname = (String) SubstrateObjectConstant.asObject(libnameNode.asConstant());
-                    if (libname != null && registeredLibraries.putIfAbsent(libname, Boolean.TRUE) != Boolean.TRUE) {
-                        /*
-                         * If a library is in our list of static standard libraries, add the library
-                         * to the linker command.
-                         */
-                        if (NativeLibrarySupport.singleton().isPreregisteredBuiltinLibrary(libname)) {
-                            nativeLibraries.addStaticJniLibrary(libname);
-                        }
-                    }
+                    registerLibrary((String) SubstrateObjectConstant.asObject(libnameNode.asConstant()));
                 }
                 /* We never want to do any actual intrinsification, process the original invoke. */
                 return false;
             }
         });
     }
 
+    void registerLibrary(String libname) {
+        if (libname != null && registeredLibraries.putIfAbsent(libname, Boolean.TRUE) != Boolean.TRUE) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [RC: Suspicious reference comparison of Boolean values (RC_REF_COMPARISON_BAD_PRACTICE_BOOLEAN)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rc-suspicious-reference-comparison-of-boolean-values-rc-ref-comparison-bad-practice-boolean).
This method compares two Boolean values using the == or != operator. Normally, there are only two Boolean values (Boolean.TRUE and Boolean.FALSE), but it is possible to create other Boolean objects using the new Boolean(b) constructor. It is best to avoid such objects, but if they do exist, then checking Boolean objects for equality using == or != will give results than are different than you would get using .equals(...).

Reply:
I'd say it's a false positive, but it's also easy to fix, so I'll see to piggyback this on some future PR:
```suggestion
        if (libname != null && registeredLibraries.putIfAbsent(libname, Boolean.TRUE) == null) {
```

Reply:
noticed, both can get the same result.